src/main/kotlin/me/partlysanestudios/partlysaneskies/features/farming/garden/CompostValue.kt

```diff
@@ -0,0 +1,267 @@
+        var matcher: Matcher
+        for (line in loreList) {
+            val unformattedLine = line.removeColorCodes()
+            matcher = pattern.matcher(unformattedLine)
+            if (matcher.find()) {
+                costLine = unformattedLine
+                break
+            }
+        }
+        matcher = pattern.matcher(costLine)
+        return if (matcher.find()) {
+            matcher.group(1)
+        } else {
+            "4000"
+        }
+    }
+
+    private fun getOrganicMatterFillLevel(inventory: IInventory): Double {
+        val organicMatterFillLevelString = getOrganicMatterFillLevelString(inventory)
+
+        return if (organicMatterFillLevelString.isEmpty()) {
+            0.0
+        } else {
+            organicMatterFillLevelString.parseAbbreviatedNumber()
+        }
+    }
+
+    private fun getOrganicMatterFillLevelString(composterInventory: IInventory): String {
+        val infoItem = composterInventory.getStackInSlot(46)
+        val loreList = infoItem.getLore()
+        var costLine = "0"
+        val regex = "\\b(\\d{1,3}(,\\d{3})*(\\.\\d+)?|\\d+k)\\b"
+        val pattern = Pattern.compile(regex)
+        var matcher: Matcher
+        for (line in loreList) {
+            val unformattedLine = line.removeColorCodes()
+            matcher = pattern.matcher(unformattedLine)
+            if (matcher.find()) {
+                costLine = unformattedLine
+                break
+            }
+        }
+        matcher = pattern.matcher(costLine)
+        return if (matcher.find()) {
+            matcher.group(1)
+        } else {
+            "0"
+        }
+    }
+
+    private fun getOrganicMatterLimit(inventory: IInventory): Double {
+        return getOrganicMatterLimitString(inventory).parseAbbreviatedNumber()
+    }
+
+    private fun getMaxCompostAbleToMake(): Double {
+        return maxCompost / compostCost
+    }
+
+    private fun getCurrentCompostAbleToMake(): Double {
+        return (maxCompost - fillLevel) / compostCost
+    }
+
+    private fun getOrganicMatterLimitString(composterInventory: IInventory): String {
+        val infoItem = composterInventory.getStackInSlot(46)
+        val loreList = infoItem.getLore()
+        var amountLine = "0/40k"
+        val regex = "(.*?)/(.*?)"
+        val pattern = Pattern.compile(regex)
+        for (line in loreList) {
+            val unformattedLine = line.removeColorCodes()
+            val matcher = pattern.matcher(unformattedLine)
+            if (matcher.find()) {
+                amountLine = unformattedLine
+                break
+            }
+        }
+        amountLine = amountLine.removeColorCodes()
+        amountLine = stripLeading(amountLine)
+        amountLine = stripTrailing(amountLine)
+        val indexOfStart = amountLine.indexOf("/")
+        amountLine = amountLine.substring(indexOfStart + 1)
+        return amountLine
+    }
+
+    private fun sortMap(map: HashMap<String, Double>): Map<String, Double> {
```

Comment:
Make this a utils method (I've seen it at least twice now)
Also maybe make it a HashMap<T, Number>.sortMap() so it sorts any type of HashMap with the second value being a number

Reply:
One of the sort maps is not like the other

Reply:
```kt
private fun <T> HashMap<T, Number>.sortMap(reversed: Boolean = false): Map<T, Number> {
    return if (reversed) {
        entries.sortedByDescending { it.value }
    } else {
        entries.sortedBy { it.value }
    }.toMap()
}
```
this should work

Reply:
```kt
fun <K : Any,V: Number> Map<K, V>.sortMap(reverseOrder: Boolean = false): Map<K, V> {
    return if (reverseOrder) {
        this.entries.sortedByDescending { it.value.toDouble() }.associate { it.toPair() }
    } else {
        this.entries.sortedBy { it.value.toDouble() }.associate { it.toPair() }
    }
}
```
this should work: [Playground](https://pl.kotl.in/mrnev4dCE)